adapters/repos/db/migrator.go

```diff
@@ -345,16 +345,22 @@ func (m *Migrator) UpdateTenants(ctx context.Context, class *models.Class, updat
 		for _, name := range updatesCold {
 			name := name
 			eg.Go(func() error {
-				idx.shardCreateLocks.Lock(name)
-				defer idx.shardCreateLocks.Unlock(name)
+				shard := func() ShardLike {
+					idx.shardInUseLocks.Lock(name)
+					defer idx.shardInUseLocks.Unlock(name)
 
-				shard, ok := idx.shards.Swap(name, nil) // swap shard for nil
-				idx.shards.LoadAndDelete(name)          // then remove entry
+					return idx.shards.Load(name)
+				}()
 
-				if !ok || shard == nil {
+				if shard == nil {
```

Comment:
not 100% sure if it's the equivalent behaviour, as at the time `if shard == nil` is evaluated no lock is being acquired so the shard could have changed its state meantime

Reply:
resolve to merge